scripts/DialogData.py

```diff
@@ -14,44 +14,193 @@
+            # make sure the nodeName is remembered within the domain
+            #if node_name not in self._nodes[domain_name]:
+            if node_name not in self._domains[domain_name]:
+                    self._domains[domain_name].append(node_name)
+
+        # Update nodes structure, if node is not there yet - add it
+        if node_name not in self._nodes:
+            self._nodes[node_name] = NodeData()
+            return self._nodes[node_name]
+        else:
+            return None
+
+    def getNode(self, node_name):
+        """  @:returns node of a given name or None
+        """
+        if node_name not in self._nodes:
+            return None
+        return self._nodes[node_name]
+
+    def getAllNodes(self):
+        """  @:returns list of all nodes
+        """
+        return self._nodes
+
+    def updateReferencesNodes(self):
+        """
+            Replaces labels by known target node_names
+            This is implementation of the second path through the dialogData.
+        """
+
+        for node_name in self._nodes: #for each node in the domain
+            nodeData = self.getNode(node_name)
+            if nodeData and nodeData._jumptoTarget:
+                label = nodeData._jumptoTarget
+                if label in self._labelsMap:
+                    node_target= self._labelsMap[label]
+                    nodeData._jumptoTarget= node_target
+                    printf('INFO: Resolving cross reference label:%s -> node_name:%s)\n', label, node_name)
+                else:
+                    printf('INFO: Label:%s not resolve, expecting tahat label is external node_name\n', label)
+
+#   DOMAINS
+#******************************************
+    def getAllDomains(self):
         return self._domains
+
+# createUnique ..
+#******************************************
+    def createUniqueIntentName(self, intent_name):
+        """
+            Creates unique intent_name based on given string
+              intent_name is stripped from not allowed characters, spaces are replaced by _
+              if the result exists a modifier is added at the end of the string
+
+              :returns unique intent_name or None if not able to create
+        """
+        #Normalize the string
+        unaccented_name=unicodedata.normalize('NFKD', intent_name).encode('ASCII', 'ignore')  # remove accents
+        unique_intent_name = toIntentName( self._NAME_POLICY, None, unaccented_name).decode('utf-8')
+        if unique_intent_name not in self._intents:
+            return unique_intent_name
+        #try to modify by a number
+        for modifier in range(0, 10000):
+            new_unique_intent_name=unique_intent_name+repr(modifier)  #create a modified one
+            # Check if the name exists
+            if new_unique_intent_name not in self._intents:
+                return new_unique_intent_name
+        return None
+
+    def createUniqueEntityName(self, entity_name):
```

Comment:
As the previous function - please merge it with `toEntityName` function:  https://github.com/IBM/watson-assistant-workbench/blob/7d164be90adae79aebd6fdd0740d9885ad57c3cb/scripts/wawCommons.py#L131

Reply:
As discussed, the part of stripping accents was based on toIntentName, the rest left for further refactoring